q2_vsearch/_cluster_features.py

```diff
@@ -177,4 +295,61 @@ def cluster_features_closed_reference(sequences: DNAFASTAFormat,
                            axis='observation',
                            include_collapsed_metadata=False)
 
-    return table, unmatched_seqs
+    return table, matched_seqs, unmatched_seqs
+
+
+def cluster_features_open_reference(ctx, sequences, table, reference_sequences,
+                                    perc_identity, strand='plus', threads=1):
+
+    cluster_features_closed_reference = ctx.get_action(
+        'vsearch', 'cluster_features_closed_reference')
+    filter_features = ctx.get_action('feature_table', 'filter_features')
+    cluster_features_de_novo = ctx.get_action(
+        'vsearch', 'cluster_features_de_novo')
+    merge = ctx.get_action('feature_table', 'merge')
+    merge_seq_data = ctx.get_action('feature_table', 'merge_seq_data')
+
+    skipped_closed_ref = True
+    try:
+        closed_ref_table, rep_seqs, unmatched_seqs = \
+            cluster_features_closed_reference(
+                sequences=sequences, table=table,
+                reference_sequences=reference_sequences,
+                perc_identity=perc_identity,
+                strand=strand, threads=threads)
+        skipped_closed_ref = False
+    except VSearchError:  # No matches
+        pass
+
+    # If cluster_features_closed_reference fails to match, we need to
+    # pass the source data into cluster_features_de_novo wholesale.
+    if skipped_closed_ref:
+        unmatched_seqs, closed_ref_table = sequences, table
+
+    # It is possible that all of the sequences matched the reference database,
+    # if that is the case, don't worry about running cluster_features_de_novo.
+    if unmatched_seqs.view(pd.Series).size:
+        unmatched_seqs_md = Metadata.from_artifact(unmatched_seqs)
+        unmatched_table, = filter_features(table=table,
+                                           metadata=unmatched_seqs_md)
+
+        de_novo_table, de_novo_seqs = cluster_features_de_novo(
+            sequences=unmatched_seqs, table=unmatched_table,
+            perc_identity=perc_identity, threads=threads)
+
+        if skipped_closed_ref:
+            outputs = (de_novo_table, de_novo_seqs, de_novo_seqs)
```

Comment:
This should still include:

```
merged_reference_seqs, = merge_seq_data(data1=reference_sequences, data2=de_novo_seqs)
```

(Same issue I commented about on the test code.)